### PR TITLE
Support TorchNet save load with NNFrames

### PR DESCRIPTION
**Pull request by hhbyyh**
support save/load torchnet model in NNFrames.

Since Torchnet does not support saveModule, we need to provide special logic for it.

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala [hunk 1]
```diff
@@ -827,22 +828,36 @@ object NNModel extends MLReadable[NNModel[_]] {
...
+        val (modulePath, weightPath) =
+          new Path(path, "module").toString -> new Path(path, "weight").toString
+        typeTag match {
+          case "TensorDouble" =>
+            ModuleLoader.loadFromFile[Double](modulePath, weightPath)
+          case "TensorFloat" =>
+            ModuleLoader.loadFromFile[Float](modulePath, weightPath)
+          case _ =>
+            throw new Exception("Only support float and double for now")
+        }
```
Comment (qiuxin2012) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: `case _` for modelType

Comment (hhbyyh) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala]: Thanks. Updated

**Comment by hhbyyh** [comment]
A gentle ping @qiuxin2012 other comments? Thanks

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/view/ZOO-PR-PIPELINE/job/ZOO-PR-Validation/2643/ passed.

**Pull request merged**